File: python/credit_check.py
import functools as ft

def credit_check(param):
    account_identifier = [int(x) for x in param]  # for every character in the string, convert to and int and append to new list
   
    for x in range(0, len(account_identifier), 2):  # for every other number in the list, multiply it by 2
        account_identifier[x]*=2

    str_ls = [str(y) for y in account_identifier]

    final_ls = []

    for z in str_ls:
        if(len(z) > 1):
            a = int(z[0]) + int(z[1])
            final_ls.append(a)
        else:
            final_ls.append(z)
            
    final_final_ls = [int(a) for a in final_ls]

    sum = ft.reduce(lambda agg, num : agg + num, final_final_ls)

    if(sum%10 == 0):
        return ("The number is valid!") 
    else:
        return("The number is invalid!")
            


# print(credit_check('5541808923795240'))

File: js/creditCheck.js
exports.creditCheck = function(param) {

    accountIdentifier = param.split('');    // Split the string into an array of numbers
    summedDigitsOverTen = [];
    count = 0;

    // Loops through array. Every other number is doubled. If the doubled number is a double digit,
    // replace it with the sum of its digits instead. push all of those numbers to a new array:
    // summedDigitsOverTen.
    for(let n of accountIdentifier) {
        if(count%2 == 0){
            if(String(((+n)*2)).length > 1) {
                let num = String(((+n)*2));
                summedDigitsOverTen.push(+num[0]+ +num[1]);
             } else {
                 summedDigitsOverTen.push(+n*2);
             }
        } else {
            summedDigitsOverTen.push(+n);
        }
        count++;
    }

    // Using the reduce, we can get the sum of all of the numbers in the array
    let sum = summedDigitsOverTen.reduce((agg, num) => agg + num);

    if(sum%10 == 0) {
        return "The number is valid!"
    } else {
        return "The number is invalid!"
    }
}